src/esp/sensor/Base.h

```diff
@@ -0,0 +1,30 @@
+// Copyright (c) Facebook, Inc. and its affiliates.
+// This source code is licensed under the MIT license found in the
+// LICENSE file in the root directory of this source tree.
+
+#pragma once
+
+#include "esp/io/json.h"
+#include "esp/scene/SceneNode.h"
+
+#include <string>
+
+namespace esp {
+namespace sensor {
+class BaseSensor : public Magnum::SceneGraph::AbstractFeature3D {
+ public:
+  explicit BaseSensor(scene::SceneNode& node, const esp::io::JsonDocument spec);
```

Comment:
I am not sure if a `esp::io::JsonDocument` will work long term.  I think having a BaseSensorSpec class and then sensors implement their own would be better.  Documenting the json document will be a nightmare and you don't get the inherent type safety/compile time errors when using a struct/class.

Reply:
I second you on this. Since the PR is WIP, I was going to iteratively refine this. The current plan is that the SensorFactory takes in a JSONDocument creates a specific SensorSpec object based on the type of sensor that is getting created. The final Sensor classes will only interact with SensorSpec objects.

Reply:
Thank you for all of the comments. :)

Reply:
@erikwijmans I think for python and javascript it will be easier for users to deal with a json sensor spec than having to allocate an object and deal with a class hierarchy.